public/js/index.js

```diff
@@ -21,7 +21,12 @@ postForm.on('submit', function (e) {
 
   var reader = new FileReader();
   reader.onload = function (evt) {
-    var vrp = JSON.parse(evt.target.result);
+    var vrp;
+    try {
+      vrp = JSON.parse(evt.target.result);
+    } catch(e) {
+      alert("Le fichier fourni n'est pas dans un format JSON valide :\n" + e);
+    }
```

Comment:
Do you know if we need `return false` here like the if condition couple of lines down below or not? 

I don't know much JS. If it is not needed here because of a generic reason (for example `alert` already cuts the execution and returns false automatically) then we can fix the if condition below. 

If it is needed then we can return here because otherwise vrp becomes an uninitialized variable (again with my zero JS knowledge, I can't see what would happen :D).

Maybe these questions are easier to address for @AntoineCAMILLE 

```suggestion
      alert("Le fichier fourni n'est pas dans un format JSON valide :\n" + e);
      return false;
    }

Reply:
I think that it could be useful to add a `return false` after the alert to avoid (maybe?) a second alert message with `"Fichier invalide"` because vrp is not initialized